docs/DeveloperGuide.md

```diff
@@ -1,9 +1,263 @@
+The user wishes to see the list of resources for all projects. He executes the following command to see them.  
+
+>list-all
+
+**Design Consideration**  
+
+Aspect: How list-all executes
+* Alternative 1 (current choice):
+  * This list-all feature will do a loop through all the projects in the projects arraylist. For each project, it will loop through all the resources and print it out.
+* Alternative 2 (none)
+
+The following sequence diagram shows how the list-all operation works:
+![add_png](puml_img/list-allfinal.png)
+
+### List all the resources for a particular project feature
+**Proposed Implementation**  
+
+Given below is an example usage scenario and how the list-all mechanism behaves at each step.  
+
+
+**Step 1**:
+The user launches the application for the first time. The application’s initialised database will be empty at the start.
+
+**Step 2**:
+The user executes the following command in order to add a certain project’s resources.
+> add p/IT3011 url/www.apple.com  
+
+> add p/CZ2003 url/www.google.com  
+
+**Step 3**:
+The user wishes to see the list of resources for one project, CZ2003. He executes the following command to see them.  
+
+>list *CZ2003*
+
+**Design Consideration**  
+
+Aspect: How list PROJECTNAME executes  
+* Alternative 1 (current choice):
+    * This feature will do a check to see if the project exists first. If it exists, it will do a loop through all the resources for this project and print it out.
+    * **Pros**: When there is an error, we will know if the error is due to the fact that the project does not exist.
+* Alternative 2:
+    * Print the project’s resources straight away without checking if project exists.  
+    * **Cons**: Harder to pinpoint the cause of the error if an error is thrown.
+
+The following sequence diagram shows how the list PROJECTNAME operation works:
+![add_png](puml_img/listPROJECTNAMEfinal.png)
+
+### Deleting resource from a specified project 
+**Proposed Implementation**
+
+Given below is an example usage scenario and how deleting resource(s) works.
+
+**Step 1**:
+The user launches the application. The application state is initialised by the Storage class which reads the text file.
+
+**Step 2**:
+The user executes the following command to delete the 2nd resource in the resource list of the project 'CS2113'.
+>delete p/CS2113 i/2
+
+**Step 3**:
+If PROJECT_NAME is not found in the database, the application prompts the user that the project is not found. 
+<br> If INDEX is invalid <i>(i.e. negative integer or exceeds the number of resources in that project)</i>, the application prompts the user to reenter a valid index.
+<br> Else, the system feedbacks to the user that the specified resource has been successfully deleted.
+
+**Step 4**:
+The user can execute `list p/CS2113` to verify that the specified resource has been deleted from the project.
+
+The following sequence diagram shows how the delete operation works:
+![delete_puml](puml_img/deleteResource.png)
```

Comment:
<img width="699" alt="dgReview2" src="https://user-images.githubusercontent.com/77370767/113388379-b5eacd00-93c0-11eb-89bb-240c5a3fb3d4.PNG">
Change this to a bigger diagram as the font is too small in the diagram and it affects readability

Reply:
I agree with Sarrah that this diagram is too wide and can't be read without zooming in.